Kuadrant/kuadrant-operator
# Reconcile Sub Component Authorino CR

Part of: #307 
RFC: https://github.com/Kuadrant/architecture/blob/main/rfcs/0006-kuadrant_sub_components_configurations.md

These changes allow a user to configure settings in the Authorino CR from the Kuadrant CR.

# Verification 

* Deploy this branch using `make local-setup`.
* Wait for deployment to complete and all resources to be a ready state.
* Deploy the custom resources.
  * `kubectl apply -f https://raw.githubusercontent.com/Boomatang/k8s_configs/kuadrant-operator-PR393/manifest/kuadrant_spec_authorino.yaml`
  * The custom manifest will deploy all the resources required for testing including an instance of Jaeger for OpenTelemetry. Jaeger needs a number of resources set up so the deployment will fail on the first run but rerun the command again to get all resources installed.
  * The Kuadrant CR has every [spec](https://github.com/Boomatang/k8s_configs/blob/kuadrant-operator-PR393/manifest/kuadrant_spec_authorino.yaml#L15681-L15687) that this PR allows configuring set. 
  * The manifest is linked to a tag [kuadrant-operator-PR393](https://github.com/Boomatang/k8s_configs/tree/kuadrant-operator-PR393) to ensure state does not change.
* Wait for the deployment to complete
* The spec that is being reconciled are listed in the RFC: [Authorino Spec](https://github.com/Kuadrant/architecture/blob/main/rfcs/0006-kuadrant_sub_components_configurations.md#authorino-spec)
* Editing the spec directly in the Authorino CR should be reverted to the spec in the Kuadrant CR
* Editing the spec in the Kuadrant CR should update the spec in the Authorino CR
  * NOTE: fully removing a spec from the kuadrant CR will not remove it from the Authorino CR. This is done to allow upgrades. A user may have custom spec already in the Limitador CR. This change only overrides the Authorino CR spec if that spec is defined in Kuadrant CR.
